inst/include/common/fims_vector.hpp

```diff
@@ -0,0 +1,483 @@
+    
+    /**
+     * @brief Returns an const iterator to the element following the last element of the vector.
+     */
+    const_iterator end() const {
+        return this->vec_m.end();
+    }
+    
+    /**
+     * @brief Returns a reverse iterator to the first element of the reversed vector. It corresponds to the last element of the non-reversed vector.
+     */
+    reverse_iterator rbegin() {
+        return this->vec_m.rbegin();
+    }
+    
+    /**
+     * @brief Returns a reverse iterator to the element following the last element of the reversed vector. It corresponds to the element preceding the first element of the non-reversed vector.
+     */
+    reverse_iterator rend() {
+        return this->vec_m.rend();
+    }
+    
+    
+    /**
+     * @brief Returns a constant reverse iterator to the first element of the reversed vector. It corresponds to the last element of the non-reversed vector.
+     */
+    const_reverse_iterator rbegin() const {
+        return this->vec_m.rbegin();
+    }
+    
+    
+    /**
+     * @brief Returns a constant reverse iterator to the element following the last element of the reversed vector. It corresponds to the element preceding the first element of the non-reversed vector.
+     */
+    const_reverse_iterator rend() const {
+        return this->vec_m.rend();
+    }
+    
+    //capacity
+    
+    
+    /**
+     * @brief Checks whether the container is empty.
+     */
+    bool empty() {
+        return this->vec_m.empty();
+    }
+    
+    /**
+     * @brief Returns the number of elements.
+     */
+    size_type size() const {
+        return this->vec_m.size();
+    }
+    
+    /**
+     * @brief Returns the maximum possible number of elements.
+     */
+    size_type max_size() const {
+        return this->vec_m.max_size();
+    }
+    
+    /**
+     * @brief Reserves storage.
+     */
+    void reserve(size_type cap) {
+        this->vec_m.reserve(cap);
+    }
+    
+    
+    /**
+     * @brief Returns the number of elements that can be held in currently allocated storage.
+     */
+    size_type capacity() {
+        return this->vec_m.capacity();
+    }
+    
+    /**
+     *  @brief Reduces memory usage by freeing unused memory.
+     */
+    void shrink_to_fit() {
+        this->vec_m.shrink_to_fit();
+    }
+    
+    //modifiers
+    
+    /**
+     * @brief Clears the contents.
+     */
+    void clear() {
+        this->vec_m.clear();
+    }
+    
+    
+    /**
+     * @brief Inserts value before pos.
+     */
+    iterator insert(const_iterator pos, const Type& value) {
+        return this->vec_m.insert(pos, value);
+    }
+    
+    /**
+     * @brief Inserts count copies of the value before pos.
+     */
+    iterator insert(const_iterator pos, size_type count, const Type& value) {
+        return this->vec_m.insert(pos, count, value);
+    }
+    
+    /**
+     * @brief Inserts elements from range [first, last) before pos.
+     */
+    template< class InputIt >
```

Comment:
what is "InputIt" and how often would we expect developers to use this insert() method?

Reply:
"InputIt" is the input iterator type.